### PR TITLE
java: Add func.yaml's options fields as metadata to labels

### PR DESCRIPTION
**Pull request by bbtong**
## Which issue(s) this PR fixes

Fixes: #69, #76

## What this PR solves

Updates Java buildpack to store metadata labels containing information about resource limits

## Details for release notes

```release-note
Updates Java buildpack to store metadata labels containing information about resource limits

```

## Describe testing done for PR

Manual, automated testing

## Special notes for your reviewer

The inner workings are a bit complex due to Java not being able to pick up `func.yaml` in the Build phase, and due to us not being on Golang 1.18 for proper generics.

I had to work around Detect phase having metadata, then unparsing the generic interface, as we have no control over the knfn or libcnb type imports. `convertLabels` does the bulk of the gross conversions.

```
"options-resources-limits": "{\"CPU\":\"1000m\",\"Memory\":\"256Mi\",\"Concurrency\":100}",
"options-resources-requests": "{\"CPU\":\"100m\",\"Memory\":\"128Mi\"}",
"options-scale": "null"
```
This results in the experience to our end user grabbing our key-value pairs of information to be identical between Python and Java.

### CODE REVIEW

**Comment by bbtong** [comment]
I'm not seeing `func.yaml` picked up at all by the Java flag 🤔 Draft until that's debugged...

**Comment by bbtong** [comment]
Debugging progress, it looks like the Java layers aren't copying over `func.yaml` so it's essentially getting discarded. We'll open an issue that will get fixed -- so this is blocked by that for now.

File: buildpacks/java/java/detect.go [hunk 1]
```diff
@@ -4,13 +4,15 @@
...
 import (
+	"encoding/json"
 	"fmt"
 	"os"
 	"path/filepath"
 
 	"github.com/buildpacks/libcnb"
 	"github.com/paketo-buildpacks/libpak"
 	"github.com/paketo-buildpacks/libpak/bard"
+	function "knative.dev/kn-plugin-func"
```
Comment (andrew-su) [inline buildpacks/java/java/detect.go:15]: This is already imported with alias `knfn`

File: buildpacks/java/java/detect.go [hunk 2]
```diff
@@ -87,3 +95,59 @@ func (d Detect) Detect(context libcnb.DetectContext) (libcnb.DetectResult, error
...
+	}
+
+	f, err := knfn.NewFunction(appPath)
+	if err != nil {
+		d.Logger.Bodyf("unable to parse '%s': %v", knfn.ConfigFile, err)
+		return []libcnb.Label{}
+	}
+	labels := d.optionsToLabels(f.Options)
+	for _, l := range labels {
+		f.Labels = append(f.Labels, function.Label{
```
Comment (andrew-su) [inline buildpacks/java/java/detect.go:114]: Use `knfn.Label` or remove and update `knfn` import

File: buildpacks/java/java/build.go [hunk 3]
```diff
@@ -110,3 +113,21 @@ func findPath(path string, r *regexp.Regexp) (string, error) {
...
+		pairs := []string{}
+		for _, val := range mapOfLabels {
+			pairs = append(pairs, val.(string))
+		}
+		labels = append(labels,
+			libcnb.Label{
+				Key:   pairs[0],
+				Value: pairs[1],
+			})
+		pairs = nil
```
Comment (andrew-su) [inline buildpacks/java/java/build.go:130]: Don't think this is needed, since you're using `pairs := []string{}` on line 121

**Review by andrew-su — CHANGES_REQUESTED** [review]

File: buildpacks/java/go.mod [hunk 4]
```diff
@@ -19,4 +19,8 @@ require (
 	golang.org/x/sys v0.0.0-20220614162138-6c1b26c55098 // indirect
 	google.golang.org/genproto v0.0.0-20220614165028-45ed7f3ff16e // indirect
 	knative.dev/kn-plugin-func v0.19.0
+
+	github.com/go-git/go-git/v5 v5.4.2 // indirect
+	github.com/xanzy/ssh-agent v0.3.0 // indirect
+	github.com/ProtonMail/go-crypto v0.0.0-20210428141323-04723f9f07d7 // indirect
```
Comment (andrew-su) [inline buildpacks/java/go.mod:25]: Might not need these, remove them and try `go mod tidy` and then rerun the builds, it should work.

Comment (bbtong) [reply buildpacks/java/go.mod]: Cleaned up!

**Pull request merged**